### PR TITLE
FINERACT-1724: Refactoring SQLInjectionValidator

### PR DESCRIPTION
**Pull request by woonys**
## Description

[FINERACT-1724] Minor bug-fixes and enhancements of 1.9.0: Refactoring SQLInjectionValidator
This PR introduces a refactoring of the SQLInjectionValidator class to improve code readability, maintainability, and adherence to the DRY (Don't Repeat Yourself) principle.

- **Introduction of Functional Interface**: A functional interface SQLCommandCondition has been introduced.
- **Refactoring of Validation Methods**: The methods validateSQLInput, validateAdhocQuery, and validateDynamicQuery have been refactored to use the new functional interface and lambda expressions.
- **Creation of Helper Methods**: Two helper methods, validateSQLCommand and validateSQLCommands, have been created to further reduce code duplication.
- **Preservation of Existing Functionality**: All existing functionality has been preserved. The refactoring focuses on improving the structure and readability of the code, not on changing its behavior.

**P.S.** I already requested jira account to make a ticket via self-serve port, but there's no approve for that request. 
image

If you guys have the authority to approve, could you please do it? #:bow: I'm a software engineer in the fintech industry and we use Fineract as our core banking platform, so I'm very interested in continuing to contribute to this open source project.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by adamsaghy** [comment]
@woonys In the meantime feel free to use FINERACT-1724 for this PR.

**Comment by woonys** [comment]
> @woonys In the meantime feel free to use [FINERACT-1724](https://issues.apache.org/jira/browse/FINERACT-1724) for this PR.

Thanks for reply, @adamsaghy. I'll modify this PR to use FINERACT-1724. 🙇  By the way, I don't have permission to add PRs to Jira issues, so can I just change the title to FINERACT-1724? Or is there something else I need to do? If you have the authority to add this PR to Jira Issue, could I ask you to do that? 🙇

**Comment by adamsaghy** [comment]
@woonys Please rebase this PR with the latest changes of the develop branch. There was a fix applied to resolve a dependency issue which blocked to execute the integrations tests (the reason why the above checks were failing)

**Comment by woonys** [comment]
@adamsaghy Thanks for notification. I just rebase this PR. Could you please check it? 🙇 

* There's an issue with integration test even though I already rebased the develop PR. Is there something I had a mistake?

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java [hunk 1]
```diff
@@ -180,4 +143,17 @@ private static void patternMatchSqlInjection(String sqlSearch, String lowerCaseS
         }
     }
 
+    private static void validateSQLCommand(String lowerCaseSQL, String[] commands, SQLCommandCondition SQLCondition) {
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java]: please rename `SQLCondition` to match with variable naming conventions. (ex.: condition)

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java [hunk 2]
```diff
@@ -180,4 +143,17 @@ private static void patternMatchSqlInjection(String sqlSearch, String lowerCaseS
...
+        for (String command : commands) {
+            if (SQLCondition.checkCondition(command, lowerCaseSQL)) {
+                throw new SQLInjectionException();
+            }
+        }
+    }
+
+    private static void validateSQLCommands(String lowerCaseSQL, List<String[]> commandsList, SQLCommandCondition conditions) {
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java]: Please rename `conditions` hence it is just one condition. (ex: condition)

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLCommandCondition.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package org.apache.fineract.infrastructure.security.utils;
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLCommandCondition.java:19]: Please add the ASF license! (you can copy from an another file)

Comment (woonys) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLCommandCondition.java:19]: @adamsaghy Thanks for review. Please check the changes. 🙇

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java (hunk 2 above)
Comment (woonys) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java]: @adamsaghy Yes, you're right. I changed it. Please check 🙇

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java (hunk 1 above)
Comment (woonys) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java]: @adamsaghy Thanks for review. I modified the name. Please check 🙇

**Comment by adamsaghy** [comment]
@woonys Changes looks good, but please squash your commits!

**Review by adamsaghy — APPROVED** [review]
LGTM

**Comment by woonys** [comment]
> LGTM

Thanks :) @adamsaghy By the way, I think I have no authority to merge. Could you merge it, please? all test have passed 🙇

**Pull request merged**